From f6e8c81eff0200e468ce2b66f053f9d952e0a91f Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Fri, 5 May 2023 05:33:54 +0200
Subject: [PATCH 2/3] gbm: Do not override pixel format set on command line

In case the framebuffer pixel format is set on command line using
the --pixelformat switch, do not rewrite it unconditionally to
ARGB8888 as that makes the --pixelformat switch unusable. Instead,
check whether pixelformat has been set already, and if so, do not
override it anymore.

Upstream-Status: Submitted [https://github.com/ardera/flutter-pi/pull/326]
Signed-off-by: Marek Vasut <marex@denx.de>
---
 src/flutter-pi.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/flutter-pi.c b/src/flutter-pi.c
index 8d3b4d6..659706b 100644
--- a/src/flutter-pi.c
+++ b/src/flutter-pi.c
@@ -1597,9 +1597,10 @@ static int init_display(void) {
      * GBM INITIALIZATION *
      **********************/
     flutterpi.gbm.device = gbm_create_device(flutterpi.drm.drmdev->fd);
-    flutterpi.gbm.format = DRM_FORMAT_ARGB8888;
     flutterpi.gbm.surface = NULL;
     flutterpi.gbm.modifier = DRM_FORMAT_MOD_LINEAR;
+    if (!flutterpi.gbm.format)
+        flutterpi.gbm.format = DRM_FORMAT_ARGB8888;
 
     flutterpi.gbm.surface = gbm_surface_create_with_modifiers(flutterpi.gbm.device, flutterpi.display.width, flutterpi.display.height, flutterpi.gbm.format, &flutterpi.gbm.modifier, 1);
     if (flutterpi.gbm.surface == NULL) {
-- 
2.39.2

